src/types.ts

```diff
@@ -145,6 +145,24 @@ export const withoutUndefinedValues = <T, K extends keyof T>(obj: T): T => {
   }, {} as T);
 };
 
+/**
+ * Return an object or an array filtering out keys that point to null values.
+ */
+export const withoutNullValues = <T, K extends keyof T>(input: T): T => {
+  if (Array.isArray(input)) {
+    return (input.map(withoutNullValues) as unknown) as T;
+  } else if (isObject(input)) {
+    return Object.keys(input)
+      .filter(key => input[key as K] !== null)
+      .reduce(
+        (acc, k) => ({ ...acc, [k]: withoutNullValues(input[k as K]) }),
+        {} as T
+      );
+  } else {
+    return input;
+  }
+};
```

Comment:
```suggestion
export const withoutNullValues = <T, K extends keyof T>(input: T): T =>
  pipe(
    input,
    E.fromPredicate(Array.isArray, () => input),
    E.map(RA.map(withoutNullValues)),
    E.map(inputAsArray => (inputAsArray as unknown) as T),
    E.orElse(
      flow(
        E.fromPredicate(isObject, () => input),
        E.map(Object.keys),
        E.map(RA.filter(key => input[key as K] !== null)),
        E.map(
          RA.reduce({} as T, (acc, k) => ({
            ...acc,
            [k]: withoutNullValues(input[k as K])
          }))
        )
      )
    ),
    E.toUnion
  );
```

We can make it more "functional" writing something like this, but I'm not sure which version is more readable

Reply:
Sorry for my late reply @fabriziopapi and thanks for this alternative version. IMHO, I would stick with the first version (the one less functional), just to be coherent with the _parent_ method, `withoutUndefinedValues`, unless there is the idea to refactor the code and rewrite it in a more functional way. 

I also want to note that this PR is related to #395 which fixes a bug and introduces a refactor to the method.